install_subrepos.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from subprocess import check_output
+
+from importlib_metadata import PackageNotFoundError, distribution
+from packaging.requirements import Requirement
+
+DEVPATH = Path(__file__).resolve().parent
+DEPS_PATH = DEVPATH / 'external-deps'
+BASE_COMMAND = [sys.executable, '-m', 'pip', 'install', '--no-deps']
+
+REPOS = {}
+for p in DEPS_PATH.iterdir():
+    if p.name.startswith('.') or not p.is_dir():
+        continue
+    try:
+        dist = distribution(p.name)._path
+    except PackageNotFoundError:
+        dist = None
+
+    REPOS[p.name] = {
+        'repo': p, 'dist': dist, 'editable': p == dist or p in dist.parents}
```

Comment:
```suggestion
        dist = None
        editable = None
    else:
        editable = (p == dist or p in dist.parents)

    REPOS[p.name] = {'repo': p, 'dist': dist, 'editable': editable}
```

My mistake here. This fixes the error when package is not installed, cleans up structure and use `None` value for `editable` in that case,